internal/cli/cmd/version.go

```diff
@@ -32,6 +37,61 @@ var (
 	Version string
 )
 
+type CiliumVersion struct {
+	client *k8s.Client
+}
+
+func NewCiliumVersion() *CiliumVersion {
+	return &CiliumVersion{client: k8sClient}
+}
+
+func (c *CiliumVersion) GetRunningCiliumVersion(ctx context.Context) (string, error) {
+	nss, err := c.client.ListNamespaces(ctx, metav1.ListOptions{})
+	if err != nil {
+		return "", fmt.Errorf("unable to list k8s namespace: %w", err)
+	}
+
+	// First look for ns kube-system
+	version, err := c.getCiliumVersionOfNamespace(ctx, "kube-system")
+	if version != "" {
+		return version, nil
+	}
+
+	if len(nss.Items) > 0 {
+		for _, namespace := range nss.Items {
+			version, err := c.getCiliumVersionOfNamespace(ctx, namespace.Name)
+			if version != "" {
+				return version, nil
+			}
+			if err != nil {
+				return "", err
+			}
+		}
+	}
+
+	return "", errors.New("unable to obtain cilium version: no cilium pods found")
+}
+
+func (c *CiliumVersion) getCiliumVersionOfNamespace(ctx context.Context, namespace string) (string, error) {
```

Comment:
Why are you duplicating
 https://github.com/cilium/cilium-cli/blob/6032e9e660af7da127163ad3e731e62ffd274120/internal/k8s/client.go#L676

here? Is there a reason we cannot use it?

Reply:
yeah, we can use it ,but have to encapsulate it ( for we don't know which namespace cilium-agent is running in)